Over21/controllers/ViewController.swift

```diff
@@ -382,3 +412,99 @@ extension ViewController: CaptureHelperDeviceDecodedDataDelegate {
     
 }
 
+extension ViewController: UPOSDeviceControlDelegate {
+    
+    func initUPOS() {
+        printerCon = UPOSPrinterController()
+        printerList = UPOSPrinters()
+        
+        //printerCon?.setLogLevel(LOG_SHOW_NEVER)
+        printerCon?.delegate = self
+        printerCon?.setTextEncoding(String.Encoding.ascii.rawValue)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(didBTStart(notification:)), name: NSNotification.Name(rawValue: __NOTIFICATION_NAME_BT_WILL_LOOKUP_), object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(didBTDeviceList(notification:)), name: NSNotification.Name(rawValue: __NOTIFICATION_NAME_BT_FOUND_PRINTER_), object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(didBTComplete(notification:)), name: NSNotification.Name(rawValue: __NOTIFICATION_NAME_BT_LOOKUP_COMPLETE_), object: nil)
+    }
+    
+    func btLookup() {
+        printerCon?.refreshBTLookup()
+    }
+    
+    @objc func didBTStart(notification: NSNotification) {
+        
+    }
+    
+    @objc func didBTComplete(notification: NSNotification) {
+        
+    }
+    
+    @objc func didBTDeviceList(notification: NSNotification) {
+        if let _ = notification.userInfo {
+            if let lookupDevice = notification.userInfo?[__NOTIFICATION_NAME_BT_FOUND_PRINTER_] as! UPOSPrinter? {
+                printerList?.addDevice(lookupDevice)
+                printerList?.save()
+                connect()
+            }
+        }
+    }
+    
+    func statusUpdateEvent(_ status: NSNumber!) {
+        
+    }
+    
+    func connect() {
+        let target = printerList?.getList()?.last as? UPOSPrinter
+        if target == nil {
+            return
+        }
+        
+        printerDevice = target
+        
+        var result = printerCon?.open(target?.modelName)
+        if result == nil || result! != 0 {
+            return
+        }
+        
+        result = printerCon?.claim(5000)
+        if result == nil || result != 0 {
+            return
```

Comment:
shouldn't we display an error if we can't claim the printer?

Reply:
added error handling and message show code